### DISCUSSION TITLE
sw_bn254 emulated overflow

### DISCUSSION THREAD

**Discussion by LautaroLasorsa** [Consensys/gnark#717] [Q&A]
Hello everyone, I'm having an error using the function ScalarMul with sw_emulated

I have the following code in the function define of my circuit:
`cr, _ := sw_emulated.New[emulated.BN254Fp, emulated.BN254Fr](api, sw_emulated.GetCurveParams[emulated.BN254Fp]())`
`field, _ := emulated.NewField[emulated.BN254Fr](api)`
`alfa := field.One() `
`P1 = cr.ScalarMul(P1, alfa)`

And when I try to test it I got the following error: 

> _ERR parsing circuit error="trying to reduce a constant, which happen to have 
> an overflow flag set\nfrontend.parseCircuit.func2\n\tcompile.go:118\nemulated.(*Field[...]).Reduce\n\tfield_ops.go:224\nemulated.(*Field[...]).IsZero\n\tfield_assert.go:197\nsw_emulated.(*Curve[...]).AddUnified\n\tpoint.go:175\nsw_emulated.(*Curve[...]).ScalarMul\n\tpoint.go:413\nCircuito.(*Circuit).Define\n\tCircuito.go:39\n"_

How can I solve it?

By the way, I deleted the cr.ScalarMul line and the test ran OK but I got the following error:

> ERR Augmented commitment hint not replaced. Proof will not be sound and verification will fail!

How can I avoid it? Does this error mean that the Circuit is useless? 

Thanks in advance.

**Comment by ivokub** [comment]
Can you post full minimal example with the witness? It looks like you are doing scalar multiplication with constant one. It may be an edge case not covered. Does this also happen on other inputs?

You are seeing the error when you only run the solver. We use challenge derived from the witness and proving key. But when you run the solver only (without proving), then it doesn't have access to proving key and uses a dummy value instead. It should be good when you run `groth16.Prove` instead of `Solve`.

**Reply by LautaroLasorsa** [reply]
Hello, I'm using the assert functions:

> 		var testigo Circuit = Circuit{
> 			Sig:  Sig,
> 			Hm:   Hm,
> 			H:    H,
> 			G:    G,
> 			Pk1:  Pk1,
> 			Apk:  Apk,
> 			Alfa: Alfa,
> 		} // testigo = witness in spanish
> 		assert.SolvingSucceeded(&circuito, &testigo, test.WithCurves(ecc.BN254), test.WithBackends(backend.GROTH16))

May this cause the problem?
